sigmastate/src/main/scala/sigmastate/UncheckedTree.scala

```diff
@@ -13,6 +15,9 @@ case object NoProof extends UncheckedTree
 
 sealed trait UncheckedSigmaTree extends UncheckedTree {
   val challenge: Array[Byte]
+  lazy val challengeOptimizedHash =
+    if (challenge.size > 4) Ints.fromByteArray(challenge)
+    else Arrays.hashCode(challenge)
```

Comment:
Why strictly > 4? Also using lazy may be problematic, because challenge is declared as val and may be overriden. This makes initialization order tricky. 
Please change to def.

Reply:
It comes from guava documentation: throws IllegalArgumentException - if bytes has fewer than 4 elements

Reply:
what about 4 bytes?

Reply:
Tested with 4 bytes, guava doesn't throw exception. See updated code in new version.